### DISCUSSION TITLE
Displaylink

### DISCUSSION THREAD

**Discussion by Zambieslar** [hyprwm/Hyprland#5911] [Q&A] [answered]
I'll be the first one to say that Displaylink is a pain in the ass. Given that Hyprland is moving away from wlroots (thank god) will we be getting display link support?

**Comment by vaxerski** [comment, marked as answer]
god knows

**Reply by Zambieslar** [reply]
I take that as a maybe? Also, un-ironically, I thought that was a very good response to my question

**Reply by James-Randall-14** [reply]
Ok, I might be totally misremembering, but I swear to god I've gotten DisplayLink to work with Hyprland in the past. It was incredibly buggy. Am I cuckoo or does it already have extremely partial support?

**Reply by Zambieslar** [reply]
Eh, kind of? I'm running a very specific version with a display link patch atm. Honestly it works pretty good and I haven't really had any issues with it. Would like to be on latest someday...

**Comment by nktnet1** [comment]
The Displaylink/Evdi patch below has been merged into Aquamarine (Hyprland is now independent of Wlroots):
- https://github.com/hyprwm/aquamarine/pull/25

It is working for me and many others, although I believe the official stance is that it is not "officially" supported.

**Comment by a-usr** [comment]
Just stumbled over this, and I have been sucessfully using DP even before the Aquamarine switch?

**Reply by nktnet1** [reply]
There was an attempted patch to the old wlroots-hyprland:

- https://github.com/hyprwm/wlroots-hyprland/pull/15

which would've solved it for some users, but not all.

More details about that here:
- https://github.com/hyprwm/wlroots-hyprland/issues/14

**Reply by a-usr** [reply]
Thank you, I was a bit confused there!

**Reply by archer-65** [reply]
Even though it works now (and before with wlroots patch), the CPU consumption is constantly quite high for me 😞

**Reply by nktnet1** [reply]
@archer-65 try turning on debug:overlay and see if your issue is similar to the one here:
- https://github.com/hyprwm/Hyprland/issues/7282

But if possible, I'd highly recommend moving away from DisplayLink in favour of DP Alt Mode / Thunderbolt 4.
Have a read through here:
- https://bbs.archlinux.org/viewtopic.php?id=265600

**Reply by archer-65** [reply]
> @archer-65 try turning on debug:overlay and see if your issue is similar to the one here:
> - https://github.com/hyprwm/Hyprland/issues/7282
> 
> But if possible, I'd highly recommend moving away from DisplayLink in favour of DP Alt Mode / Thunderbolt 4.
> Have a read through here:
> - https://bbs.archlinux.org/viewtopic.php?id=265600

Yes! Same issue. FPS range between 200-300 with FPS capped...

Unfortunately DisplayLink is my only option. My work laptop is the Dell Vostro 3520 variant without USB C.

**Reply by nktnet1** [reply]
That's probably it - you should perhaps follow-up in that issue and hope it'll gain some traction.

Other things to try would be testing DisplayLink on Gnome/Kde (ensuring that they're running on Wayland) to see if you run into the same CPU issue - if so, it may not be a Hyprland issue. Apart from DisplayLink just not being a good piece of software, I would suspect there's also some problems with Wayland handling evdi drivers, especially for dynamic contents like YouTube videos (I don't recall seeing the issue on Xorg or Windows 11).

I've since switched to TB4 and haven't looked back.

---

P.s. specific to YouTube, turning off Ambient Mode could help (although more so for the GPU):
- https://github.com/hyprwm/Hyprland/issues/6935

Relevant DisplayLink thread with high CPU utilisation:
- https://www.displaylink.org/forum/showthread.php?t=67355

Info on DisplayLink vs Alt Mode (DisplayPort or Thunderbolt):
- https://kb.plugable.com/docking-stations-and-video/chipset-modal
- https://juicedsystems.com/blogs/news/displaylink-vs-thunderbolt-4-understanding-the-key-differences?srsltid=AfmBOoppuxeuz2M01ofN71O46THjarzc7aYtXqNyE-lu1eO9KAl6ACPi

**Reply by archer-65** [reply]
> That's probably it - you should perhaps follow-up in that issue and hope it'll gain some traction.
> 
> Other things to try would be testing DisplayLink on Gnome/Kde (ensuring that they're running on Wayland) to see if you run into the same CPU issue - if so, it may not be a Hyprland issue. Apart from DisplayLink just not being a good piece of software, I would suspect there's also some problems with Wayland handling evdi drivers, especially for dynamic contents like YouTube videos (I don't recall seeing the issue on Xorg or Windows 11).
> 
> I've since switched to TB4 and haven't looked back.
> 
> ---
> 
> P.s. specific to YouTube, turning off Ambient Mode could help (although more so for the GPU):
> - https://github.com/hyprwm/Hyprland/issues/6935
> 
> Relevant DisplayLink thread with high CPU utilisation:
> - https://www.displaylink.org/forum/showthread.php?t=67355
> 
> Info on DisplayLink vs Alt Mode (DisplayPort or Thunderbolt):
> - https://kb.plugable.com/docking-stations-and-video/chipset-modal
> - https://juicedsystems.com/blogs/news/displaylink-vs-thunderbolt-4-understanding-the-key-differences?srsltid=AfmBOoppuxeuz2M01ofN71O46THjarzc7aYtXqNyE-lu1eO9KAl6ACPi
> 
> 
> 

I'm following the issue, thank you!

GNOME is "fine". Lately I've had problems with artifacts and annoying glitches (reported on the EVDI GitHub repo), but the CPU usage is much better (0.3% usage on GNOME, 17% on Hyprland and Sway with patched wlroots 😔)

**Reply by nktnet1** [reply]
Great! I'm assuming this:
- https://github.com/DisplayLink/evdi/issues/489

I recalled someone having a similar issue in the past on Nvidia drivers (graphical artifacts):
- https://www.displaylink.org/forum/showthread.php?p=88202

This leads to this guide showing how to add EVDI to kernel when system starts
-https://support.displaylink.com/knowledgebase/articles/1843660-screen-freezes-after-opening-an-application-only

Unsure if it's related to what you're facing on GNOME Wayland so I didn't reply there, but worth a shot. Could also end up fixing other issues as a side-effect.
